invenio_app_ils/search_permissions.py

```diff
@@ -53,15 +53,17 @@ def search_filter_record_permissions():
     return dsl.Q("bool", filter=[combined_filter])
 
 
-def ils_search_factory(self, search, validator=None):
-    """Search factory with Query String validator.
+def ils_search_factory(self, search, query_parser=None, validator=None):
```

Comment:
is there any reason why you didn't add the new kwarg at the end of the function signature? I am worried that it will break the usages if it is passed implicitly as argument not kwarg

Reply:
Yes, because we are passing this function to the config `search_factory_imp` where `invenio_records_rest` passes the query parser/modifier as the 2nd param.
Ref: https://github.com/inveniosoftware/invenio-records-rest/pull/329/files